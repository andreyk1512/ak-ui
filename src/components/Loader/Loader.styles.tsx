import { css, SerializedStyles } from '@emotion/react';
import * as styles from '../../styles';
import { LoaderSize } from './Loader.types';

const sizeMap: Record<LoaderSize, SerializedStyles> = {
  default: css`
    width: 40px;
    height: 40px;
  `,
  small: css`
    width: 24px;
    height: 24px;
  `,
};

export const spinnerStyles = (size: LoaderSize) => css`
  animation: spin 1s linear infinite;
  color: ${styles.color.lightSeaGreen};

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }

  ${sizeMap[size]}
`;

export const wrapperStyles = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 10px;
  color: ${styles.color.yankeesBlue};
`;
