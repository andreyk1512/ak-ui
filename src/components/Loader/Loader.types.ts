export type LoaderSize = 'small' | 'default';

export type LoaderProps = {
  size?: LoaderSize;
  label?: string;
}
