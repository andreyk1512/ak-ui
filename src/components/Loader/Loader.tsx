import { Typography } from '../Typography';
import { LoaderProps } from './Loader.types';
import LoaderIcon from '../../assets/icons/loader.svg?react';
import { spinnerStyles, wrapperStyles } from './Loader.styles';

export const Loader = ({ size = 'default', label }: LoaderProps) => {
  if (label) {
    return (
      <div css={wrapperStyles}>
        <LoaderIcon aria-label="loader" css={spinnerStyles(size)} />
        <Typography.Text tiny={size === 'small'}>{label}</Typography.Text>
      </div>
    );
  }

  return <LoaderIcon aria-label="loader" css={spinnerStyles(size)} />;
};
