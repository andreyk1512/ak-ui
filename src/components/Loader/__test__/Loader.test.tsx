import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Loader } from '../Loader';

describe('Loader', () => {
  it('should render loader component', () => {
    const { container } = render(<Loader />);
    const loader = screen.getByLabelText('loader');

    expect(loader).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render loader component with label', () => {
    const label = 'Loading...';
    const { container } = render(<Loader label={label} />);
    const loader = screen.getByLabelText('loader');
    const labelComponent = screen.getByText(label);

    expect(loader).toBeInTheDocument();
    expect(labelComponent).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
