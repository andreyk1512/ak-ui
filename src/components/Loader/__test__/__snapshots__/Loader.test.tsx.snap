// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render loader component 1`] = `
<span
  aria-label="loader"
  class="css-5ftavm"
/>
`;

exports[`Loader should render loader component with label 1`] = `
<div
  class="css-zclaep"
>
  <span
    aria-label="loader"
    class="css-5ftavm"
  />
  <span
    class="css-1ti22to"
  >
    Loading...
  </span>
</div>
`;
